runtime/lua/vim/treesitter/languagetree.lua

```diff
@@ -280,32 +337,22 @@ end
 --- Note: This call invalidates the tree and requires it to be parsed again.
 ---
 ---@private
----@param regions table List of regions this tree should manage and parse.
-function LanguageTree:set_included_regions(regions)
```

Comment:
A couple of questions:

First, is `set_included_regions` public API or not? It has `@private` luadoc so it isn't in the published docs, but it seems that it used to be. The `@private` was added in #20142. As I was writing this I thought I was free to change the signature and I think I might have to backpedal that.

If it is a public API...

Since we're normalising `Range4`s, that means both the setter and the getter have some undocumented functionality. I'd guess that most people would interpret these APIs as documented as a simple getter and setter combo. But that's already not how it works because we are normalising `Range4`s in-place. So `set_included_regions` currently accepts range4 + tsnode, and `included_regions` returns range6 + tsnode. It looks as though range4 support was added in order to implement the `#offset!` directive, and range6's accidentally leaked out (only on child langs). Out [in the wild][] one or two people seem to be using it a little, entirely with tsnodes and assuming tsnodes so far.

There are a couple of options to rectify this:

* normalise everything to range6 and document that you can supply range4s or range6s when setting and that `included_regions` will return range6 always;
* stop normalising in-place, make it behave like a getter/setter for the root, document that _range4_ works for the setter, pretend range6 in the public API never happened and then, for child langs, re-normalise the return value of `included_regions` back to range4.
* some other permutations of those options

Any ideas? In any case at minimum I will probably normalise tsnodes to range6 internally. We need range6 so we can arbitrarily split them and have a convenient comparison value (the buffer byte offset), and we need uniformity lest the clipping code get really messy,.

[in the wild]: https://github.com/search?type=code&q=set_included_regions+OR+included_regions+language%3Alua+path%3A%2F%5Elua%2F

Reply:
> It has `@private` luadoc so it isn't in the published docs

that means it's private

Reply:
I was mainly confused because it didn't have an _underscore prefix like the other private methods. Thanks for clarifying.